mathics/core/pattern.py

```diff
@@ -289,27 +289,28 @@ def per_expr(yield_expr, expr_groups, sum=0):
 
                         if expr_groups:
                             expr, count = expr_groups.popitem()
-                            max_per_pattern = count / len(patterns)
+                            max_per_pattern = int(count / len(patterns))
```

Comment:
Here is an interesting thing about Python 2.7 vs Python 3.7 (or maybe just Python 3+):

    Python 2.7.14 (default, Sep 22 2017, 00:06:07)
    >>> 5 / 2
    2
    >>> type(5 / 2)
    <type 'int'>

    Python 3.7.0 (default, Aug  4 2018, 02:33:39)
    >>> 5 / 2
    2.5
    >>> type(5 / 2)
    <class 'float'>

Reply:
yeah, i got that one right away 👍